src/resources/at-date.js

```diff
@@ -0,0 +1,76 @@
+// @ts-check
+
+class AtDate extends HTMLElement {
+  constructor() {
+    super();
+  }
+
+  connectedCallback() {
+    const innerText = this.innerText || this.textContent;
+    const dateText = innerText?.trim().slice(4) ?? "";
+
+    const date = new Date(dateText);
+
+    const formatter = new Intl.DateTimeFormat(undefined, {
+      dateStyle: "long",
+      timeStyle: "long",
+    });
+
+    const shadow = this.attachShadow({ mode: "open" });
+
+    const time = document.createElement("time");
+    time.setAttribute("datetime", dateText);
+    const slot = document.createElement("slot");
+
+    time.appendChild(slot);
+    shadow.appendChild(time);
+
+    const stylesheet = new CSSStyleSheet();
+
+    stylesheet.replace(`
+      :host {
```

Comment:
why can't we put this on `styles.css`?

Reply:
cause it's attached to the custom element - the shadow dom means you can't (easily) write css in the root element that penetrates into custom elements. also cause i'm setting `content` prop on `:host:after` based on component props

Reply:
fuckd up...